### PR TITLE
Setup firebase ml model downloader directories.

### PR DESCRIPTION
**Pull request by annzimmer**
Ready for initial review.

### CODE REVIEW

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloader.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.firebase.ml.modeldownloader;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import com.google.android.gms.common.internal.Preconditions;
+import com.google.firebase.FirebaseApp;
+
+public class ModelDownloader {
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloader.java:21]: Usually main entry points to the SDK have `Firebase` as prefix, i.e. `FirebaseModelDownloader`. We might have missed this in API review, worth being consistent though. pls check with the council

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.List;
+
+public class ModelDownloaderRegistrar implements ComponentRegistrar {
+
+  @Override
+  @NonNull
+  public List<Component<?>> getComponents() {
+    return Arrays.asList(
+        Component.builder(ModelDownloader.class)
+            .add(Dependency.required(FirebaseApp.class))
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java:33]: If you only need access to `FirebaseOptions`, consider depending on them directly.

```suggestion
            .add(Dependency.required(FirebaseOptions.class))
```

Context: https://github.com/google/guice/wiki/InjectOnlyDirectDependencies

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import androidx.annotation.NonNull;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.components.Component;
+import com.google.firebase.components.ComponentRegistrar;
+import com.google.firebase.components.Dependency;
+import com.google.firebase.platforminfo.LibraryVersionComponent;
+import java.util.Arrays;
+import java.util.List;
+
+public class ModelDownloaderRegistrar implements ComponentRegistrar {
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java:26]: please add a `@hide` javadoc comment to this class.

Comment (annzimmer) [reply firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java]: done

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java (hunk 2 above)
Comment (annzimmer) [reply firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloaderRegistrar.java]: Updated but will likley need firebaseApp by the time the code is fully written. But I can update as needed.

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloader.java (hunk 1 above)
Comment (annzimmer) [reply firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/ModelDownloader.java]: reaching out to api team - will likely go with FirebaseMlModelDownloader

File: firebase-ml-modeldownloader/src/lint.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (annzimmer) [reply firebase-ml-modeldownloader/src/lint.xml]: done

**Comment by vkryachko** [comment]
I think the api.txt is still incorrect, you can automatically fix it by running `./gradlew :firebase-ml-modeldownloader:generateApiTxtFile`

File: firebase-ml-modeldownloader/firebase-ml-modeldownloader.gradle [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+plugins {
+    id 'firebase-library'
+}
+
+firebaseLibrary {
+    testLab.enabled = true
```
Comment (vkryachko) [inline firebase-ml-modeldownloader/firebase-ml-modeldownloader.gradle:20]: You should change this to `false` until you actually add integration tests. Testlab considers absence of tests as a failure.

**Review by vkryachko — APPROVED** [review]

File: firebase-ml-modeldownloader/firebase-ml-modeldownloader.gradle (hunk 5 above)
Comment (annzimmer) [reply firebase-ml-modeldownloader/firebase-ml-modeldownloader.gradle]: done in previous submit

**Pull request merged**